### PR TITLE
implement glue code for query and queryRecord using pouchdb-find

### PR DESCRIPTION
**Pull request by BernardTolosajr**
Usage:
MyModel.query({selector: { name: "Mario" }, sort: ["name"]})
MyModel.query({selector: {debut: {$gte: 1990}})
MyModel.query({selector: { series: "Mario" }, sort: [{series: "desc"}, {debut: "desc"}]})
MyModel.queryRecord({selector: { name: "Mario" }})

### CODE REVIEW

**Comment by aexmachina** [comment]
@BernardTolosajr and @nolanlawson, can I get an update on the status of this PR please. I'm doing a tidy-up pass through our PRs.

**Comment by nolanlawson** [comment]
I'm not really maintaining ember-pouch anymore because of a lack of time. As for pouchdb-find compatibility with relational-pouch/ember-pouch I'd say I'm very skeptical that it could work well although this PR looks to me like about the best possible way to do it.

My only major feedback is that I had kind of hoped pouchdb-find functionality could be rolled into relational-pouch rather than ember-pouch, but as I said I basically have zero time for this project anymore, so my opinion probably matters very little at this point. 😅

**Comment by aexmachina** [comment]
@BernardTolosajr I think that @nolanlawson raises a good point here. This functionality probably belongs in relational-pouch. Are you at all interested in adding it there? @nolanlawson are there many users of relational-pouch outside ember-pouch? @broerse what do you think about adding this functionality into ember-pouch?

**Comment by BernardTolosajr** [comment]
@aexmachina Sure, ill check it.

**Comment by broerse** [comment]
@BernardTolosajr @aexmachina I think because ember-pouch is the "interface" for ember-data this functionality needs to be in this module.  If pouchdb-find is merged into relational-pouch or something ember-pouch still needs to facilitate this. So perhaps lets merge this.

**Comment by nolanlawson** [comment]
> @nolanlawson are there many users of relational-pouch outside ember-pouch?

@aexmachina npm says 325 downloads in the past month for `relational-pouch` vs 866 for `ember-pouch` (not even sure how that's possible? maybe transitive deps aren't counted?) but yeah, corresponds roughly to my feeling, which is that ember-pouch is about 3x more popular than relational-pouch.

In any case, my goal with relational-pouch was to isolate the "pouch-y" stuff into that repo, and the "ember-y" stuff into this repo. In my mind, pouchdb-find falls squarely in the "pouch-y" part, hence my comment.

OTOH maybe it makes sense to have ember-pouch just directly depend on relational-pouch and pouchdb-find. I'm fine with that too, if folks feel that it solves a real problem and isn't an unwieldy maintenance burden.

**Comment by broerse** [comment]
@BernardTolosajr I like to merge this. Can you fix the conflicts?

**Comment by BernardTolosajr** [comment]
@broerse fixed conflicts.

**Comment by fsmanuel** [comment]
@BernardTolosajr 👍  I really like what you did but can we change `selector` to `filter` to be closer to the JSON::API Spec and make it easier for people to change adapters as they need?

**Comment by broerse** [comment]
@BernardTolosajr I also think changing `selector` to `filter` is better.

**Comment by BernardTolosajr** [comment]
@fsmanuel Thanks! yup filter is better. @broerse :+1:

File: addon/adapters/pouch.js [hunk 1]
```diff
@@ -229,10 +281,43 @@ export default DS.RESTAdapter.extend({
...
+        var results = {};
+
+        var rows = payload.docs.map((row) => {
+          var parsedId = db.rel.parseDocID(row._id);
+          if (!Ember.isEmpty(parsedId.id)) {
+            row.data.id = parsedId.id;
+            return row.data;
+          }
+        });
+
```
Comment (aexmachina) [inline addon/adapters/pouch.js]: This will yield undefined elements in the array if there's no id. Is this what you want? Maybe you should use reduce?

Comment (BernardTolosajr) [inline addon/adapters/pouch.js]: My bad i did not anticipate missing id here, ill check on this using reduce.

**Comment by broerse** [comment]
@BernardTolosajr @aexmachina Good catch. Fix this in a new PR. Add test?

**Comment by BernardTolosajr** [comment]
@broerse :thumbsup: ill submit a new PR.

**Comment by BernardTolosajr** [comment]
@aexmachina @broerse  upon initial testing for missing id found out that pouchdb auto-generate the _id if its not provided.

**Comment by broerse** [comment]
@aexmachina @BernardTolosajr Nice! I wil release 4.0.0 then. Thanks!

**Comment by aexmachina** [comment]
@BernardTolosajr okay cool, but I still think that the mapper should be updated. Also, upgrading to v4.0.0 breaks the app because `bower_components/pouchdb-find/dist/pouchdb.find.js` is not installed. Maybe we should include upgrade instructions to the changelog?

**Comment by broerse** [comment]
@aexmachina no it was some typo junk.  v4.0.1  fixes this

**Comment by aexmachina** [comment]
I'm running v4.0.1 and `ember serve` fails unless I add pouchdb-find to bower.json.

**Comment by broerse** [comment]
@aexmachina but you did not run `ember install ember-pouch` ;-)

**Comment by aexmachina** [comment]
@broerse okay, that is the right way to upgrade an ember addon isn't it :)

**Pull request merged**